pkg/epp/scheduling/config.go

```diff
@@ -52,3 +59,58 @@ var defaultConfig = &SchedulerConfig{
 	picker:              defPlugin,
 	postSchedulePlugins: []plugins.PostSchedule{},
 }
+
+func CreateConfig(opts ...ConfigOption) *SchedulerConfig {
+	config := &SchedulerConfig{
+		preSchedulePlugins:  []plugins.PreSchedule{},
+		postSchedulePlugins: []plugins.PostSchedule{},
+		scorers:             map[plugins.Scorer]int{},
+		filters:             []plugins.Filter{&sheddableRequestFilterV2{}},
+		picker:              &picker.MaxScorePicker{},
+	}
+	for _, opt := range opts {
+		opt(config)
+	}
+	return config
+}
+
+type ConfigOption func(*SchedulerConfig)
+
+func WithPrefixPlugin(prefixConfig prefix.Config) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		prefixPlugin := prefix.New(prefixConfig)
+		cfg.preSchedulePlugins = append(cfg.preSchedulePlugins, prefixPlugin)
+		cfg.postSchedulePlugins = append(cfg.postSchedulePlugins, prefixPlugin)
+		cfg.scorers[prefixPlugin] = prefixConfig.Weight
+	}
+}
+
+func WithQueuePlugin(queueConfig scorer.QueueScorerConfig) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		queuePlugin := &scorer.QueueScorer{}
+		cfg.scorers[queuePlugin] = queueConfig.Weight
+	}
+}
+
+func WithKVCachePlugin(kvCacheConfig scorer.KVCacheScorerConfig) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		kvCachePlugin := &scorer.KVCacheScorer{}
+		cfg.scorers[kvCachePlugin] = kvCacheConfig.Weight
+	}
+}
+
+type sheddableRequestFilterV2 struct{}
+
+func (p *sheddableRequestFilterV2) Name() string {
+	return "sheddableRequestFilterV2"
+}
+
+func (p *sheddableRequestFilterV2) Filter(ctx *types.SchedulingContext, pods []types.Pod) []types.Pod {
+	if ctx.Req.Critical {
+		// Allow all pods to pass through if the request is critical, even if all pods reach their capacity.
+		return pods
+	}
+
+	// Only allow pods that have enough capacity to handle the request.
+	return filter.HasCapacityFilter.Filter(ctx, pods)
+}
```

Comment:
this doesn't belong here.
it should go to filter package and be with Capital letter to allow its use from outside the filter pacakge.

Reply:
Pls review a small refactor that cleans up the "default filter" in scheduler and makes it reusable.

https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/802